### DISCUSSION TITLE
android apk cant install

### DISCUSSION THREAD

**Discussion by foxyloxy-eng** [GerryDazoo/SlingerPlayer#10] [General]
Getting you cant install the app on your device. Using a pixel 7 pro. Also tried on an ndvidia shield tube and just gets to a ?flame? Logo and hangs?

**Comment by foxyloxy-eng** [comment]
Apparently the pixel 7 will only run 64-bit apps.

**Reply by GerryDazoo** [reply]
That's a bummer. I've got tool chain issues building 64-bit apps. But if I can figure it out I'll make a 64 bit version.

**Reply by foxyloxy-eng** [reply]
Thank you

**Reply by bradut** [reply]
I think I might have the same issue.

I just tried to install the APK onto my tablet which is powered by a 64 bit CPU ([HiSilicon Kirin 659](https://www.notebookcheck.net/HiSilicon-Kirin-659-SoC-Benchmarks-and-Specs.249596.0.html)) and, after it asks for permission to access files and media, the app  displays the image below and stops.
![image](https://user-images.githubusercontent.com/2496596/232247115-50a48881-ed04-4c78-a842-7a81531e731f.png)

The player works fine on my Android phone.

**Reply by GerryDazoo** [reply]
I don't think it's a CPU issue, the tablet is probably running a old incompatible Android version.

**Reply by bradut** [reply]
The tablet is running Android 8.0

**Reply by blasco403** [reply]
I get the same flame screen when launching after install on a brand new nvidia shield pro with Android 11. It also doesn't show an icon that the app is installed. I had to launch from the app control screen where you would force stop/etc. 

When I get some time I am going to roll it back to Android 8 and see what happens.

**Reply by bradut** [reply]
In my case, the icon does show up.

I've tried installing on: 
-  Tablet (Huawei) with a 64 bits CPU, Android 8.0 
    - icon shows up, app asked for some permissions when ran first time, but I only displays the flame.
- Android TV Box (Minix) with a 64 bits CPU, Android 7.x
   - same issue as above
- Old tablet, Android 4.x (was able to play SlingPlayer) 
  - app cannot be installed (which I expected).

As I already mentioned, the player **works fine on my Android phone**, Android 11 with a **64 bits CPU**. 

Therefore, the App can work on a 64-bit CPU. 

I looked in my tablet to find a log file, but no luck, not sure where to find it.
- Found some XMLs in: mnt/sdcard/Android/data/com.embarcadero.SlingerPlayer
- Similar files found on SD card: storage/emuIated/O/Android/data/com.embarcadero.SlingerPlayer

**Reply by bradut** [reply]
UPDATE:
It was my fault, as I didn't realize that I had to keep the **tablet in portrait mode** first in order to see the remote control form.
The app runs fine !

**Comment by ooom416354** [comment]
Hi, I have the same issue where it won't install on my Pixel 7.  Any update or any workaround I can do?

**Reply by GerryDazoo** [reply]
The Pixel 7 will only run 64-bit apps and the current code is compiled as a 32-bit app. I'm unaware of  any work around. I don't plan on investing my time to get a 64 bit version running any time soon.

**Reply by ooom416354** [reply]
Thanks. Is the code available somewhere?  Maybe I'll take a stab at trying to configure.  Thanks Gerry.

**Reply by GerryDazoo** [reply]
64 bit version now available. Give it a try and let me know

**Reply by ooom416354** [reply]
Works awesome! Thanks very much Gerry.  One question, is there a way to force full screen when turning to landscape?  It stretches the video but leaves the notification bar with clock and battery, etc. again thank you!

**Reply by blasco403** [reply]
Works great on firestick and nvidia shield pro.

Thanks Gerry.

**Comment by martysm** [comment]
The 64bit compilation works great on my Motorola G-Power(2021).  Thank you Gerry!  
With auto-rotate enabled I get sorta full screen in landscape and a small screen below which is a large remote when vertically oriented which I assume is what is intended.  About one or two seconds less overall latency than VLC and no noticeable mismatch between audio and video.  Behaves well on local LAN or through WG-VPN.

Is there a file with the remote codes I can tweak somewhere?  I can change channels (up/down/by number) but I can't trigger the Comcast DVR.
![Screenshot_20231130-152734](https://github.com/GerryDazoo/SlingerPlayer/assets/79178124/884e5837-cd11-4eda-a0e2-2e5b7bc2e24b)
![Screenshot_20231130-152720](https://github.com/GerryDazoo/SlingerPlayer/assets/79178124/cdb597f3-1256-4e75-8426-752b500fbf09)

**Reply by GerryDazoo** [reply]
I was just watching the house debate like you. Yes the swapping to landscape/full screen on rotation is design intent. The IR remote codes are stored on the slingbox and the mapping of buttons to IR codes is done on the server-side.

**Reply by GerryDazoo** [reply]
Also, it looks like the remote web page is a little large for your screen. You can fix that on the server side by modifying the remote.txt file to reduce the font size for the buttons. 

style=.button {
      border: none;
      color: white;
      background-color: blue;
      padding: 0px 2px;
      text-align: center;
      text-decoration: none;
      display: inline-block;
      **font-size: 20px;**
      margin: 4px 2px;
     cursor: pointer;
    }

There is two other places.

**Reply by martysm** [reply]
Gerry - I'm just getting around to this.... what am I missing.  I edited in three places and restarted the server but no change in the button size on the player:

style=.button {
      border: none;
      color: white;
      background-color: blue;
      padding: 0px 2px;
      text-align: center;
      text-decoration: none;
      display: inline-block;
      font-size: 20px;
      margin: 4px 2px;
     cursor: pointer;
    }
    
   .round {
      border: none;
      color: white;
      background-color: green;
      padding: 0px 2px;
      text-align: center;
      text-decoration: none;
      display: inline-block;
      font-size: 20px;
      margin: 4px 2px;
     cursor: pointer;
      border-radius: 50%;
    }
        
    .text {
      border: none;
      color: black;
      background-color: lightblue;
      text-align: center;
      text-decoration: none;
      display: inline-block;
      font-size: 20px;
      margin: 4px 2px;
      cursor: pointer;
    }

**Reply by GerryDazoo** [reply]
I was just pointing to where you need to change the font size. 20 is the default for Slinger. You may need to make it smaller.
The other possibility is your not editing the file that Slinger is actually using. Check the logs on startup.
